explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+ * who the point of contact is
+ * when it was last updated
+ * what are the authoritative name servers for the domain
+ * what are the mail exchangers for the domain (i.e., which systems are responsible for email for this domain)
+ * how host names are mapped to IP addresses, and vice-versa
+ * what information about other services, such as kerberos realms, LDAP servers, etc (usually on internal domains only) are there
+ * and more
+
+When DNS was first conceived, security wasn't a top priority. At its origins, DNS is susceptible to multiple vulnerabilities, and has many weaknesses. Most of them are a consequence of spoofing: there is no guarantee that the reply you received to a DNS query a) was not tampered with; b) came from the true source.
+
+This is not news, and other mechanisms on top of DNS and around it are in place to counteract that weakness. For example, the famous *HTTPS* padlock that can be seen when accessing most websites nowadays, which uses the TLS protocol to both authenticate the website, and encrypt the connection. It doesn't prevent DNS spoofing, and your web browser might still be tricked into attempting a connection with a fraudulent website, but the moment the TLS certificate is inspected, a warning will be issued to the user. Depending on local policies, the connection might be even immediately blocked. Still, DNS spoofing is a real problem, and TLS itself is subject to other types of attacks.
+
+## What is it?
+
+DNSSEC, which stands for Domain Name System Security Extensions, is an extension to DNS that introduces digital signatures. This allows each DNS response to be verified for:
+
+ * integrity: The answer was not tampered with and did not change during transit.
+ * authenticity: The data came from the true source, and not another entity impersonating the source.
+
+It's important to note that DNSSEC, however, will NOT encrypt the data: it is still sent in the clear.
+
+DNSSEC is based on public key cryptography, meaning that every DNS zone has a public/private key pair. The private key is used to sign the zone's DNS records, and the corresponding public key can be used to verify those signatures. This public key is also published in the zone. Anyone querying the zone can also fetch the public key to verify the signature of the data.
+
+A crucial question arises: How can we trust the authenticity of this public key? The answer lies in a hierarchical signing process. The key is signed by the parent zone's key, which, in turn, is signed by its parent, and so on, until we reach the root DNS zone. The root zone keys are implicitly trusted, serving as the foundation of the trust chain. This sequence of signed keys forms the chain of trust, ensuring the integrity and authenticity of DNS records.
```

Comment:
It's better, but I think I now need to clarify that the root zone keys which are implicitly trusted, are so because they are shipped within all resolvers.